### ISSUE TITLE
Darkness areas

### ISSUE THREAD

**Issue by MisterE123** [DokimiCU/Exile#5]
See below

**Comment by MisterE123** [comment]
there are still large areas of darkness, this seems to be where the snow layer in the sky gets close to the ground. Valleys mapgen.

**Comment by MisterE123** [comment]
Sorry, first post was a mistake, a copy of the other post. Please ignore it

**Comment by DokimiCU** [comment]
This is a compatibility problem going to Minetest 5.3. Did not exist in previous versions of Minetest

So far I've seen:
- The river bottom sediment for the biome covers most of map (wet soils), but should only be in river valleys.
- snow in the sky. Occurs at heights for alpine biomes. Seems to be the "dust" nodes (i.e. snow). Snow should only be on the ground in mountains.
- ERROR[Server]: Server::ActiveObjectMgr::addActiveObjectRaw(): no free id available. Seems to be part of same problem (I've closed that thread). Shows up when going through "sky snow". Don't know why.


Clearly something has been changed in Minetest. I don't know what. I don't know what would cause these errors.

My only guess is that it is similar to the previous "Empty black world" issue. That was fixed by making map settings explicit in the code.

I tried doing the same for some mgvalleys specific settings. 

minetest.set_mapgen_setting("mg_valleys_spflags", "altitude_chill, humid_rivers, vary_river_depth, altitude_dry", true)

Didn't seem to help.
